spec.emu

```diff
@@ -874,7 +880,7 @@ contributors: Nicolò Ribaudo
           </dl>
 
           <emu-alg>
-            1. Assert: This call to Evaluate is not happening at the same time as another call to Evaluate within the surrounding agent.
+            1. Assert: This call to Evaluate is not happening at the same time as another call to Evaluate within the surrounding agent<ins>, unless it was initiated by EvaluateSync</ins>.
```

Comment:
We should probably reword this assertion to explicitly describe what it's actually needed for:
```suggestion
            1. <del>Assert: This call to Evaluate is not happening at the same time as another call to Evaluate within the surrounding agent.</del>.
            1. <ins>Assert: _module_ and all its recursive dependency have [[Status]] set to one of ~linked~, ~evaluated~, ~evaluating-async~, or ~evaluated~.</ins>
```

Reply:
Yeah that's a much nicer formal definition thanks.

Reply:
I've inverted this assertion to explicitly call out `evaluating` here, in case it might be missed in the list of states as the key invariant being maintained.